### PR TITLE
Database models

### PR DESCRIPTION
**Pull request by KarmaKamikaze**
- SourceController has been implemented
- The interface ISourceRepository and IRepository has been modified to accept error handling
- SourceRepository returns status codes on database executions

### CODE REVIEW

File: DocumentDataAPI/DocumentDataAPI/Data/Repositories/ISourceRepository.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+using DocumentDataAPI.Models;
+
+namespace DocumentDataAPI.Data.Repositories;
+
+public interface ISourceRepository
```
Comment (Sarmisuper) [inline DocumentDataAPI/DocumentDataAPI/Data/Repositories/ISourceRepository.cs]: Hvorfor ikke lade denne inherit fra IRepository?

**Review by Sarmisuper — DISMISSED** [review]
lgtm

File: DocumentDataAPI/DocumentDataAPI/Data/Repositories/ISourceRepository.cs (hunk 1 above)
Comment (KarmaKamikaze) [reply DocumentDataAPI/DocumentDataAPI/Data/Repositories/ISourceRepository.cs]: Fixed 👍

**Review by Sarmisuper — DISMISSED** [review]
lgtme

**Review by Sarmisuper — APPROVED** [review]
lgtm

**Pull request merged**